### PR TITLE
Add TM integration and separate docker-compose services for Varnish

### PR DESCRIPTION
**Pull request by AbdelrahmanElawady**
Add Traffic Monitor integration with a daemon running on Varnish cache servers to report whether Varnish is running or not and to report load average and interface stats. Also, running Varnish now with CIAB is more user friendly and does not require changing any file manually just specifying varnish docker compose file with the default one.

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Monitor
- CDN in a Box

## What is the best way to verify this PR?

After building rpms needed for CIAB:
- Run using `docker-compose -f docker-compose.yml -f docker-compose.varnish.yml up` that should run all cache servers with Varnish.
- Open Traffic Portal and navigate to Cache stats. Cache servers should be healthy.
- Stop Varnish in any cache server with `docker-compose exec edge systemctl stop varnish.service`.
- Traffic Portal will show edge as unhealthy.

## PR submission checklist
- [x] This PR has tests 
- [x] This PR has documentation 
- [ ] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by AbdelrahmanElawady** [comment]
marked draft until #7727 is reviewed

File: infrastructure/cdn-in-a-box/varnish/vstats.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	"time"
+)
+
+type app struct {
+	mu            *sync.Mutex
+	vstats        vstats
+	checkInterval time.Duration
+}
+
+type vstats struct {
```
Comment (limited) [inline infrastructure/cdn-in-a-box/varnish/vstats.go:43]: Can this struct be shared with anything that Traffic Monitor already deserializes?

Comment (AbdelrahmanElawady) [reply infrastructure/cdn-in-a-box/varnish/vstats.go:43]: It definitely should be shared however it's implemented as just "main" package separated from TC so it's not a package that TM can call. However, I think it should be a package under TC and be built with `pkg` considering statistics will be added too. Should that be in this PR or done separately?

Comment (limited) [reply infrastructure/cdn-in-a-box/varnish/vstats.go:43]: If its ok with you, let's keep it separate, since this one is pretty well baked already

Comment (AbdelrahmanElawady) [reply infrastructure/cdn-in-a-box/varnish/vstats.go:43]: Yep, will open an issue for it to be done later.

**Review by limited — APPROVED** [review]

**Pull request merged**